interaction-manager/src/lib_components/resize.js

```diff
@@ -0,0 +1,230 @@
+import { clamp } from '../utils/utility-functions';
+import actions from './actions';
+
+const hash = (Math.random() + 1).toString(36).substring(7);
+
+/**
+ * Allows you to resize elements on the screen
+ */
+class Resize {
+    /**
+     *
+     * @param {Object} options
+     * @param {string} options.element
+     * @param {number} options.edgeWidth
+     * @param {function} options.onWidthChange
+     * @param {function} options.onHeightChange
+     * @param {number} options.widthMin
+     * @param {number} options.widthMax
+     * @param {number} options.heightMin
+     * @param {number} options.heightMax
+     */
+    constructor(options) {
+        this.options = options;
+        this.options.edgeWidth = options.edgeWidth || 5;
+        this.options.heightMin = this.options.heightMin || 50;
+        this.options.heightMax = this.options.heightMax || window.innerHeight;
+        this.options.widthMin = this.options.widthMin || 50;
+        this.options.widthMax = this.options.widthMax || window.innerWidth;
+
+        this.resizableElement = null;
+        this.enabled = false;
+        this.activeEdge = null;
+        this.edges = {
+            bottom: null,
+            right: null,
+            bottomRight: null,
+        };
+
+        this.actionHeight = `resize-height-${hash}`;
+        this.actionWidth = `resize-width-${hash}`;

```

Comment:
```suggestion
        this.heightAction = `resize-height-${hash}`;
        this.widthAction= `resize-width-${hash}`;

```

the opposite order implies that they are numbers and it is confusing when we pass them to execute as an action name.